mitodl/mit-open
# fix price display and update vertical cards

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4934
Closes https://github.com/mitodl/hq/issues/4924

### Description (What does it do?)
This PR updates the logic for fetching and displaying prices on `LearningResourceCard` and its variants. Price display was added to the horizontal cards, styling and displaying the information as applicable per the designs based on the size of the card.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/d9d2edf9-8aab-4403-90dd-ecab7dff7c69)
![image](https://github.com/user-attachments/assets/95ad9cc7-40f1-47e1-99aa-9a4ff6173f46)
![image](https://github.com/user-attachments/assets/9aa92295-d04c-47dd-9d07-dd16b13851e4)
![image](https://github.com/user-attachments/assets/efbe34ae-d657-4639-9667-2dabd8996c70)
![image](https://github.com/user-attachments/assets/fa90e78d-0d2c-46e5-95cd-2361dd2f5b41)
![image](https://github.com/user-attachments/assets/6e676321-9562-4e4e-8f05-0d82cac9545b)

### How can this be tested?
 - Spin up this branch of `mit-open`
 - Ensure you have a variety of courses (free, not free, with certificate), probably best done by backpopulating OCW and MITx
 - Ensure that you are logged in and that your user has a profile with some topics selected
 - Visit the home page at http://localhost:8063/ and take a look at the medium sized cards there, ensuring that they follow the designs
 - Visit http://localhost:8063/search and browse around, changing the search filters to display different results
   - Ensure that the cards look like the designs
   - Ensure that all free courses properly display "Free" where the price would normally be
 - Visit https://localhost:8063/dashboard/ to view the small cards there, ensuring that they are consistent with the designs
